test/pydmap_tests/test_dmapread.py

```diff
@@ -0,0 +1,325 @@
+        self.assertEqual(dm_records[0]['stid'].shape[0], 14)  # this will be file dependent... future working test project.
+
+    # TODO: Again dependent on the file used :/
+    def test_integrity_check_corrupt_file1(self):
+        """
+        Test test_initial_data_integrity on a corrupt file
+
+        Expected bahaviour: raises pydmap expection
+        """
+        dmap = pydarn.DmapRead(corrupt_file1)
+        with self.assertRaises(pydarn.pydmap_exceptions.MismatchByteError):
+            dmap.test_initial_data_integrity()
+
+    def test_read_corrupt_file1(self):
+        """
+        Test read_records on a corrupt file
+
+        Expected bahaviour: raises pydmap expection
+        """
+        dmap = pydarn.DmapRead(corrupt_file1)
+        with self.assertRaises(pydarn.pydmap_exceptions.DmapDataTypeError):
+            dmap.read_records()
+
+    def test_integrity_check_corrupt_file2(self):
+        """
+        Test test_initial_data_integrity on a corrupt file
+
+        Expected bahaviour: raises pydmap expection
+        """
+        dmap = pydarn.DmapRead(corrupt_file2)
+        with self.assertRaises(pydarn.pydmap_exceptions.ZeroByteError):
+            dmap.test_initial_data_integrity()
+
+    def test_read_currupt_file2(self):
+        """
+        Test read_records on a corrupt file
+
+        Expected bahaviour: raises pydmap expection
+        """
+        dmap = pydarn.DmapRead(corrupt_file2)
+        with self.assertRaises(pydarn.pydmap_exceptions.ZeroByteError):
+            dmap.read_records()
+
+    # FIXME: have not gotten this test to work with the streaming. I believe it is an utf-8 problem
+    #def test_dmap_read_through_randomization(self):
+    #    """Randomly corrupts 5% of the data in a record and tries to parse.
+    #        Fails if an non Dmap exception is thrown as this means there is a
+    #        missing check against bad input. Increase the number of tests to
+    #        stress test."""
+
+    #    num_tests = 100
+    #    file_path = fitacf_file
+    #    dm = pydarn.DmapRead(file_path)
+    #    dm.cursor = 0
+    #    dm.read_record()
+
+    #    dmap_length = dm.cursor
+    #    dmap_to_randomize = dm.dmap_bytearr[0:dmap_length]
+
+    #    seed = int(time.time())
+    #    random.seed(seed)
+
+    #    for x in range(0, num_tests):
+    #        gc.collect()
+    #        randomizer = [os.urandom(1) if random.randint(0, 100) >= 95
+    #                      else '\x00' for j in range(0, dmap_length)]
+
+    #        corrupted_dmap = [chr(a ^ ord(b))
+    #                          for a, b in zip(dmap_to_randomize, randomizer)]
+    #        self.assertRaises(pydarn.DmapDataError,pydarn.parse_dmap_format_from_stream, corrupted_dmap)
+    #        del(records)
+    #        del(randomizer)
+    #        del(corrupted_dmap)
+
+
+if __name__ == '__main__':
+    """
+    Runs the above class in a unittest system.
+    Roughly takes 467 seconds.
```

Comment:
I wonder if its possible to use mp with unittest.

Reply:
Again not within PR scope, but you can use NOSE or the extension of unittests testtools to get concurrency. 

If you do it yourself you are asking for some inaccurate reporting because the print statements may not be thread-safe. 

Again out of scope and my test suite takes 6 seconds to run so currently not worth it for now. 

Will look into it when I focus on the testing Schema of this library.